crypto/src/poseidon_hash.rs

```diff
@@ -0,0 +1,37 @@
+// XXX move into poseidon377 crate?
```

Comment:
One thing I'm wondering is whether it would make sense to split the upstream code in `ark-sponge` into two levels, one that just does the permutation, and then another that tracks the sponge state on top of that.

Reply:
created https://github.com/arkworks-rs/sponge/issues/29